gufe/protocols/protocolunit.py

```diff
@@ -302,22 +304,30 @@ def execute(self, *,
             result = ProtocolUnitResult(
                 name=self.name, source_key=self.key, inputs=inputs, outputs=outputs
             )
-
         except Exception as e:
             if raise_error:
                 raise
 
+            outputs = {}
+
             result = ProtocolUnitFailure(
                 name=self._name,
                 source_key=self.key,
                 inputs=inputs,
-                outputs=dict(),
+                outputs=outputs,
                 exception=(e.__class__.__qualname__, e.args),
                 traceback=traceback.format_exc()
             )
 
-        # TODO: change this part once we have clearer ideas on how to inject
-        # persistent storage use
+        # extract any files from the values of the dict that _execute
+        # returns; ensure that we move these to shared so they can be used
+        # by later units
+        for label, path in outputs.items():
+            is_file = (isinstance(path, Path) and path.exists()
+                       and not path.is_dir())
+            if is_file:
+                shared.store_path(f"{self.key}/{label}", path)
```

Comment:
We discussed and didn't quite decide on if outputs should explicitly be listed to be copied up, or instead the shared dir passed to _execute can just be walked and blindly copied up. It might come down to if there's cases where items are required in shared (for archiving) but never used as inputs for future Units